src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1755,28 +1758,45 @@ public static Matrix4x4 Transform(Matrix4x4 value, Quaternion rotation)
         /// </summary>
         /// <param name="matrix">The source matrix.</param>
         /// <returns>The transposed matrix.</returns>
-        public static Matrix4x4 Transpose(Matrix4x4 matrix)
+        public static unsafe Matrix4x4 Transpose(Matrix4x4 matrix)
         {
-            Matrix4x4 result;
-
-            result.M11 = matrix.M11;
-            result.M12 = matrix.M21;
-            result.M13 = matrix.M31;
-            result.M14 = matrix.M41;
-            result.M21 = matrix.M12;
-            result.M22 = matrix.M22;
-            result.M23 = matrix.M32;
-            result.M24 = matrix.M42;
-            result.M31 = matrix.M13;
-            result.M32 = matrix.M23;
-            result.M33 = matrix.M33;
-            result.M34 = matrix.M43;
-            result.M41 = matrix.M14;
-            result.M42 = matrix.M24;
-            result.M43 = matrix.M34;
-            result.M44 = matrix.M44;
-
-            return result;
+            if (Sse.IsSupported)
+            {
+                Matrix4x4 result = default;
+                var t0 = Sse.Shuffle(Sse.LoadVector128(&matrix.M11), Sse.LoadVector128(&matrix.M21), 0x44);
+                var t1 = Sse.Shuffle(Sse.LoadVector128(&matrix.M31), Sse.LoadVector128(&matrix.M41), 0x44);
+                var t2 = Sse.Shuffle(Sse.LoadVector128(&matrix.M11), Sse.LoadVector128(&matrix.M21), 0xEE);
+                var t3 = Sse.Shuffle(Sse.LoadVector128(&matrix.M31), Sse.LoadVector128(&matrix.M41), 0xEE);
+
+                Sse.Store(&result.M11, Sse.Shuffle(t0, t1, 0x88));
```

Comment:
It would be nice to comment what `0x88` is (it looks like `ZXZX`)

The same for 0xDD (`WYWY`), 0x44 (`YXYX`), and 0xEE (`WZWZ`)

Reply:
turns out Unpack+Move works better and no magic numbers are involved 🙂